src/IECore/StreamIndexedIO.cpp

```diff
@@ -2523,23 +2586,27 @@ void StreamIndexedIO::Index::commitNodeToSubIndex( DirectoryNode *n )
 	if ( n->subindex() == DirectoryNode::NoSubIndex )
 	{
 		MemoryStreamSink sink;
-		io::filtering_ostream compressingStream;
-		compressingStream.push( io::gzip_compressor() );
-		compressingStream.push( sink );
-		assert( compressingStream.is_complete() );
+		io::filtering_ostream outIndexStream;
+		outIndexStream.push( sink );
+		assert( outIndexStream.is_complete() );
 
-		writeNodeChildren( n, compressingStream );
+		writeNodeChildren( n, outIndexStream );
 
-		compressingStream.pop();
-		compressingStream.pop();
+		outIndexStream.pop();
 
-		char *data=nullptr;
-		std::streamsize sz;
-		sink.get( data, sz );
-		uint32_t subindexSize = sz;
+		// compress the output stream using blosc before writing
+		char* indexData = nullptr;
+		std::streamsize indexDataSize;
+
+		sink.get(indexData, indexDataSize);
+
+		std::vector<char> compressedIndex;
+		compress( indexData, indexDataSize, compressedIndex, 9, "lz4", 1, BLOSC_MAX_BUFFERSIZE, 0);
```

Comment:
Should this be using `m_compressor`, `m_compressionLevel` etc? I guess the problem is that those default to no compression, but we've always had compression unconditionally on for the index?

Reply:
Yup. Do you propose we store the index without compression?

Reply:
No - I presume that would be a pretty big regression in terms of previous performance. But I do think it's inconsistent that we have parameters that seem like they would control this, but are being ignored. I guess our options are :

1. Leave it as it is.
2. Use the parameters if they have been specified, but otherwise fall back to the current behaviour.
3. Always use the parameters, but default them to having compression on.

3 has a nice consistency to it, but I think my original objection was that it was slower, do you still have the numbers on that?

Reply:
we decided to ignore this on the grounds that we never want an uncompressed index, but it is legitimate to turn off compression on the rest of the file.